ttnn/cpp/ttnn/operations/pool/generic/device/pool_multi_core_program_factory.cpp

```diff
@@ -78,6 +109,14 @@ Pool2D::MultiCore::cached_program_t pool2d_multi_core_sharded_with_halo_v2_impl_
 
     TT_FATAL(nblocks == 1, "Multiple blocks not yet supported");
 
+    TT_FATAL(
+        !(input_shape[3] > tt::constants::TILE_WIDTH && input_shape[3] % tt::constants::TILE_WIDTH != 0),
+        "channel > TILE_WIDTH and not a multiplier of TILE_WIDTH outputs incorrect results (#15731)");
+
+    TT_FATAL(
+        !(pool_type == Pool2DType::AVG_POOL2D && is_wide_reduction),
+        "avg_pool2d wide reduction outputs incorrect results (#14459)");
```

Comment:
Here I try to explicitly block known incorrect cases. Not sure if this is the preferrable way